sycl/source/detail/device_kernel_info.cpp

```diff
@@ -63,7 +63,6 @@ void DeviceKernelInfo::setCompileTimeInfoIfNeeded(
     const CompileTimeKernelInfoTy &Info) {
   if (!isCompileTimeInfoSet())
     CompileTimeKernelInfoTy::operator=(Info);
-  assert(isCompileTimeInfoSet());
```

Comment:
@sergey-semenov I removed this assert because in case of app was compiled with the 6.3 compiler, the kernel size is not passed from the headers to the runtime.

Reply:
After discussion with @sergey-semenov, we decided to keep this assert in the preview mode.
Corresponding comment is added.

Reply:
This deserves more explanation (probably in the PR description). New `KernelData` is under `source/detail` and doesn't cross ABI boundary. What are we changing here that has ABI effects and why can't that be done in a separate PR?

Reply:
But I added comment in place. And the issue happens only in case of 6.3 or 6.2 headers. There is nothing about ABI here.